src/ccsr/src/config.rs

```diff
@@ -76,11 +79,9 @@ impl ClientConfig {
         self
     }
 
-    /// Overrides the url.
-    ///
-    /// Useful for certain kinds of tunneling.
-    pub fn override_url(mut self, url: Url) -> ClientConfig {
-        self.url = url;
+    /// Adds a `Proxy` to the list of proxies that the client will use.
+    pub fn proxy(mut self, proxy: Proxy) -> ClientConfig {
```

Comment:
very nit: maybe call it `add_proxy` (previously, the could be only one override, but now we can have multiple proxies)